.ci/jobs.t/elastic+elasticsearch+periodic+snyk-dependency-monitoring.yml

```diff
@@ -0,0 +1,23 @@
+---
+- job:
+    name: elastic+elasticsearch+%BRANCH%+snyk-dependency-monitoring
+    workspace: /dev/shm/elastic+elasticsearch+%BRANCH%+snyk-dependency-monitoring
+    display-name: "elastic / elasticsearch # %BRANCH% - snyk dependency monitoring"
+    description: "Publishing of the Elasticsearch %BRANCH% dependencies graph to snyk dependency monitoring"
+    builders:
+      - inject:
+          properties-file: '.ci/java-versions.properties'
+          properties-content: |
+            JAVA_HOME=$HOME/.java/$ES_BUILD_JAVA
+            RUNTIME_JAVA_HOME=$HOME/.java/$ES_RUNTIME_JAVA
+      - shell: |
+          #!/usr/local/bin/runbld --redirect-stderr
+          set -euo pipefail
+          set +x
+          VAULT_TOKEN=$(vault write -field=token auth/approle/login role_id=$VAULT_ROLE_ID secret_id=$VAULT_SECRET_ID)
+          export VAULT_TOKEN
+          export data=$(vault read -format=json secret/elasticsearch-ci/snyk)
+          export snyk_token=$(echo $data | jq -r .data.token)
+          unset VAULT_TOKEN data
+          set -x
+          $WORKSPACE/.ci/scripts/run-gradle.sh uploadSnykDependencyGraph -PsnykToken=$snyk_token
```

Comment:
We can't do `set -x` and then include the token in the command line as this is going to leak the secret. We'll either need to tweak the build to pull this from an env var, or omit the `set -x` bit so we don't print this command. I'm inlined to do the former, as it's probably less likely that we'll make a mistake and accidentally leak the token in console output when using environment variables as we do for most other secrets.